integration-tests/src/test/java/com/datastax/oss/driver/api/core/session/GuavaSession.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2017-2017 DataStax Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.session;
+
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.internal.core.cql.DefaultPrepareRequest;
+import com.datastax.oss.driver.internal.core.session.SessionWrapper;
+import com.google.common.util.concurrent.ListenableFuture;
+
+public class GuavaSession extends SessionWrapper {
+
+  static final GenericType<ListenableFuture<AsyncResultSet>> ASYNC =
+      new GenericType<ListenableFuture<AsyncResultSet>>() {};
```

Comment:
Not that we care for this test, but for real extensions I think it will be a good practice to make the type tokens public just in case.
And also add the convenience methods via an interface.